### DISCUSSION TITLE
Change default Port 80 to (f.e.) 8082

### DISCUSSION THREAD

**Discussion by nadar** [php/frankenphp#1288] [Q&A] [answered]
I am new to FrankenPHP (and Caddie?) and I’m curious about how to change the default port from 80 (because it’s already in use on my remote development machine). I tried mapping another port, but it doesn’t seem to work:

```
docker run \
    -e FRANKENPHP_CONFIG="worker ./public/index.php" \
    -v $PWD:/app \
    -p 8081:80 -p 8082:443 -p 8082:443/udp \
    dunglas/frankenphp
```

I expected that I could then use `localhost:8081`, but it doesn’t work. Any help would be appreciated.

(As a side question, why is an SSL port required? For instance, if I build a Dockerfile and run this inside a Kubernetes cluster, the SSL configuration is handled differently. I’d like to disable SSL entirely if I don’t need it.)

**Comment by withinboredom** [comment]
You need to add a new parameter to tell frankenphp what ports it is listening on:

```
-e SERVER_NAME=":8081,https:8082" \
-p 8081:8081 -p 8082:8082
```

Or something like that. Basically, caddy is trying to redirect you to https:443 and it isn't there.

**Reply by nadar** [reply, marked as answer]
I had to fix SERVER_NAME, but you actually fixed my problem. Thank you so much @withinboredom 

```
docker run \
    -e FRANKENPHP_CONFIG="worker ./public/index.php" \
    -e SERVER_NAME=":8081 https:8082" \
    -v $PWD:/app \
    -p 8081:8081 -p 8082:8082 \
    dunglas/frankenphp
```

**Reply by acelaya** [reply]
Hey! What's the right way to do this when running FrankenPHP via static binary? It doesn't seem to take into account the `SERVER_NAME` env var.

**Reply by withinboredom** [reply]
I believe the flag is `--listen :8082` when using the `php-server` command. You can always use `--help` to see what options are available.

**Reply by acelaya** [reply]
> I believe the flag is `--listen :8082` when using the `php-server` command

Yeah, that works. I thought it didn't at first because I was passing `--listen :8080` and then I was getting an error from the browser when visiting `localhost:8080`, but I just realized the issue is that I'm getting redirected from `http://localhost:8080` to `https://localhost`.

So a different problem. `--listen` works as expected. Thanks!
